frontend/packages/console-app/src/components/tour/tour-context.ts

```diff
@@ -16,9 +16,16 @@ import {
 } from './utils';
 import { TourDataType, Step } from './type';
 
-export const tourReducer = (state: TourState, action: TourActions) => {
+const initializeState = (completed: boolean) => ({
+  completedTour: completed,
+  stepNumber: 0,
+  startTour: !completed,
+});
+export const tourReducer = (state: TourState, action: { type: TourActions; payload?: boolean }) => {
   const { stepNumber } = state;
-  switch (action) {
+  switch (action.type) {
+    case TourActions.initialize:
+      return initializeState(action.payload);
```

Comment:
no need to declare a method here. just set the state directly here
```suggestion
      return {
  completedTour: completed,
  stepNumber: 0,
  startTour: !completed,
};
```

Reply:
done